### ISSUE TITLE
Please warn about magic aliases

### ISSUE THREAD

**Issue by sitsofe** [koalaman/shellcheck#534]
Trying to predict what the following will print  and predict what the following will print out in bash:

```
alias magicalias="echo old"
type magicalias
x() { type magicalias ; magicalias; }
x
alias magicalias="echo new"
x
type x
```

This behaviour appears to be brought to you by [magic aliases](http://www.chiark.greenend.org.uk/~sgtatham/aliases.html).

(This might be a duplicate of issue #438)

**Comment by deryni** [comment]
The idea here being to warn that the alias is going to be expanded in the function definition and not inserted literally?

Also this only works in an interactive shell by default as scripts don't expand aliases at all unless `expand_aliases` is enabled in non-interactive/script context.

**Comment by sitsofe** [comment]
@deryni:
That's correct just a warning (but what do you do about the "type" case where it wasn't expanded?) that it will be expanded literally. Thanks for the comment that it only happens interactive shells - I guess the warning would have to carry that note too...

**Comment by deryni** [comment]
By "the "type" case where it wasn't expanded" do you mean how `type magicalias` stays `magicalias` in the function definition where the `magicalias` command didn't?

That's because of how aliases expand. They don't expand mid-command; only at the start.

Though I agree that this is certainly surprising behavior. I wouldn't have expected it even though I can understand it.

**Comment by sitsofe** [comment]
Re type: Yes that exactly what I meant and thanks for the explanation!
